README.md

```diff
@@ -922,6 +931,62 @@ interface INPAttribution {
 }
 ```
 
+#### `LongAnimationFrameSummary`
+
+```ts
+/**
+ * An object containing potentially-helpful debugging information summarized
+ * from the LongAnimationFrames intersecting the INP event.
+ *
+ * NOTE: Long Animation Frames below 50 milliseconds are not reported, and
+ * so their scripts cannot be included. For Long Animation Frames that are
+ * reported, only scripts above 5 milliseconds are included.
+ */
+export interface LongAnimationFrameSummary {
+  /**
+   * The number of Long Animation Frame scripts that intersect the INP event.
+   * NOTE: This may be be less than the total count of scripts in the Long
+   * Animation Frames as some scripts may occur before the interaction.
+   */
+  numScripts?: number;
+  /**
+   * The slowest Long Animation Frame script that intersects the INP event.
+   */
+  slowestScript?: PerformanceScriptTiming;
+  /**
+   * The INP phase where the longest script ran.
+   */
+  slowestScriptPhase?:
+    | 'inputDelay'
+    | 'processingDuration'
+    | 'presentationDelay';
+  /**
+   * The total blocking durations in each phase by invoker for scripts that
+   * intersect the INP event.
+   */
+  totalDurationsPerPhase?: Record<
+    'inputDelay' | 'processingDuration' | 'presentationDelay',
+    Record<string, number>
+  >;
+  /**
+   * The total forced style and layout durations as provided by Long Animation
+   * Frame scripts intercepting the INP event.
+   */
+  totalForcedStyleAndLayoutDuration?: number;
+  /**
+   * The total non-force (i.e. end-of-frame) style and layout duration from any
+   * Long Animation Frames intercepting INP event.
+   */
+  totalNonForcedStyleAndLayoutDuration?: number;
+  /**
+   * The total duration of Long Animation Frame scripts that intersect the INP
+   * duration. Note, this includes forced style and layout within those
+   * scripts.
+   */
+  totalScriptDuration?: number;
```

Comment:
As noted in https://github.com/GoogleChrome/web-vitals/issues/559#issuecomment-2524195543 I added this as I thought it might be helpful to differentiate between script and non-script (style and layout) time.

Reply:
Here's an example (more from LoAF than from INP) where the ability to split by script, style&layout (and forced Style and Layout) was thought to be useful: https://webperformance.slack.com/archives/C04BK7K1X/p1733613959706609?thread_ts=1732204038.067829&cid=C04BK7K1X